libsel4vm/src/arch/x86/boot.c

```diff
@@ -76,25 +83,119 @@ static vm_frame_t pd_alloc_iterator(uintptr_t addr, void *cookie)
     return frame_result;
 }
 
+static int make_guest_pd_continued(void *access_addr, void *vaddr, void *cookie)
+{
+    uint64_t *pd = vaddr;
+    int num_entries = BIT(seL4_PageBits) / sizeof(pd[0]);
 
-static int make_guest_page_dir(vm_t *vm)
+    /* Brute force 1:1 entries. */
+    for (int i = 0; i < num_entries; i++) {
+        /* Present, write, user, page size 2M */
+        pd[i] = (uint64_t)((uint64_t)((i) << PAGE_BITS_2M) | PAGE_ENTRY);
```

Comment:
Better cast the base type to `uint64_t`, then all operation will happen on an integer of this type
```suggestion
        pd[i] = ((uint64_t)i) << PAGE_BITS_2M) | PAGE_ENTRY;
```